### PR TITLE
Strict version on hkl for csx

### PR DESCRIPTION
**Pull request by jrmlhermitte**
We should enforce hkl>=0.3.12 from now on for CSX (and other beamlines eventually)

### CODE REVIEW

File: recipes-tag/23-id-1-csx1-collection/meta.yaml [hunk 1]
```diff
@@ -1,10 +1,10 @@
-{% set year = "17" %}
-{% set cycle = "Q1" %}
+{% set year = "2018" %}
+{% set cycle = "1" %}
```
Comment (mrakitin) [inline recipes-tag/23-id-1-csx1-collection/meta.yaml]: Should it be `2` already?

File: recipes-tag/23-id-1-csx1-collection/meta.yaml [hunk 2]
```diff
@@ -1,10 +1,10 @@
...
 {% set version ="0" %}
 
 package:
   name: 23-id-1-csx1-collection
-  version: {{ year }}{{ cycle }}.{{ version }}
+  version: {{ year }}C{{ cycle }}.{{ version }}
 
 build:
   number: 1
```
Comment (mrakitin) [inline recipes-tag/23-id-1-csx1-collection/meta.yaml]: `number: 0`?

File: recipes-tag/23-id-1-csx1-collection/meta.yaml (hunk 1 above)
Comment (jrmlhermitte) [reply recipes-tag/23-id-1-csx1-collection/meta.yaml]: thanks! updated

File: recipes-tag/23-id-1-csx1-collection/meta.yaml (hunk 2 above)
Comment (jrmlhermitte) [reply recipes-tag/23-id-1-csx1-collection/meta.yaml]: thanks! updated

**Pull request merged**